### ISSUE TITLE
Color themes not working

### ISSUE THREAD

**Issue by ManuGraiph** [GothenburgBitFactory/taskwarrior#3461]
Hi,

For some reason, i can't change any color output either in my taskwarrior or taskwarrior-tui gui when sourcing any theme file in the taskrc. I'm not sure if i'm missing an option, other than the `include file` for the theme.

I'm on arch, my terminal is wezterm and my term is xterm-256color.

Any idea what could it be?

**Comment by djmitche** [comment]
Are you just seeing plain text? @adriangalilea has been working with themes and might be able to help debug.

**Comment by ManuGraiph** [comment]
Yep, the colors are always the same, regardless of what "theme" i select. If i copy the color lines in the taskrc, it's the same.

**Comment by adriangalilea** [comment]
https://gist.github.com/adriangalilea/c0756755d364e87ccca186b1e0361d3d

This is the theme I came up with, that I save on  `~/.task/themes` and then on  `~/.taskrc` I add: `include ~/.task/themes/violet.theme`

Not sure what problem are you experiencing, but you can try `task color` command to debug if the program/terminal is capable of producing the colors and determine if it's a config issue or lower level.

**Comment by celadevra** [comment]
> Yep, the colors are always the same, regardless of what "theme" i select. If i copy the color lines in the taskrc, it's the same.

In my case, I put my taskrc under `~/.config/task`, but forgot to delete `~/.taskrc`. The latter has higher precedence when TW looks for its config.

**Comment by djmitche** [comment]
You can verify that with `task diag`, which will show the config file location.

**Comment by djmitche** [comment]
Hm, this is the second report of task color configuration not "taking", so perhaps there's something funny going on here. If the correct config file is, indeed, in use, then perhaps this has to do with ordering of config within the file?

**Comment by Shadabdullah** [comment]
> Hi,
> 
> For some reason, i can't change any color output either in my taskwarrior or taskwarrior-tui gui when sourcing any theme file in the taskrc. I'm not sure if i'm missing an option, other than the `include file` for the theme.
> 
> I'm on arch, my terminal is wezterm and my term is xterm-256color.
> 
> Any idea what could it be?

I am facing same issues ,

**Comment by djmitche** [comment]
Does `task show | grep ^color` show the expected colors?

**Comment by ziofil** [comment]
I have the same issue. Here's the output of the command you just suggested:

<img width="327" alt="Screenshot 2024-09-13 at 5 25 28 PM" src="https://github.com/user-attachments/assets/2b3b0808-99ba-4eed-a95d-ae90446cb57e">

**Comment by djmitche** [comment]
Are those different from the colors in your theme? If so, where does `task diag` say it is reading the configuration from?

**Comment by ziofil** [comment]
No, they are always the same color regardless of the color theme in the settings. It also looks like `taskrc` is found and accessed correctly.

<img width="553" alt="Screenshot 2024-09-17 at 8 36 49 AM" src="https://github.com/user-attachments/assets/8bc5024c-0485-42cd-b031-8b6bc693c1f7">

**Comment by djmitche** [comment]
Thanks! So it seems like there's something related to the `include` in `/Users/fliippo/.taskrc`. Try changing that to a path that doesn't exist, and see if you get an error message?

**Comment by ziofil** [comment]
Not sure I understand: should I change the path of the data/hooks inside taskrc? Or should I temporarily move `.taskrc` elsewhere? This is my current taskrc (I have moved it to the XDG directory as suggested, but nothing has changed, btw):

```text
# [Created by task 3.1.0 9/13/2024 08:07:20]
# data.location=/Users/filippo/.task
news.version=3.1.0

# To use the default location of the XDG directories,
# move this configuration file from ~/.taskrc to ~/.config/task/taskrc and update location config as follows:

data.location=~/.local/share/task
hooks.location=~/.config/task/hooks

# Color theme (uncomment one to use)
#include light-16.theme
#include light-256.theme
#include dark-16.theme
#include dark-256.theme
include dark-red-256.theme
#include dark-green-256.theme
#include dark-blue-256.theme
#include dark-violets-256.theme
#include dark-yellow-green.theme
#include dark-gray-256.theme
#include dark-gray-blue-256.theme
#include solarized-dark-256.theme
#include solarized-light-256.theme
#include no-color.theme
```

If I move taskrc I get this error message:

```text
task show
A configuration file could not be found in /Users/filippo

Would you like a sample /Users/filippo/.taskrc created, so Taskwarrior can proceed? (yes/no) n
Cannot proceed without rc file.
```

**Comment by djmitche** [comment]
No, I meant to change
```
include dark-red-256.theme
```
to something like
```
include no-such-file.theme
```
and see if you get an error message.

**Comment by ziofil** [comment]
Sorry for replying so late. Yes I do get an error:
```
[taskwarrior-tui error]: Unable to run `task show`.

Could not find file in CWD, directory of config file or search paths 'no-such-file.theme'.

Please check your configuration or open a issue on github.

If you need additional help, please report as a github issue on https://github.com/kdheepak/taskwarrior-tui
```

**Comment by djmitche** [comment]
I guess that error is from tui, and I don't really know how that tool works. But I suppose `task` would say the same?

I'm pretty stumped, then. I don't know how Taskwarrior could be reading _most_ of the settings in a config file but not all of them. I guess I'd suggest making one last check of `.taskrc` to see if there are any `includes` that aren't commented out or anything else that might be overriding the color scheme.

**Comment by djmitche** [comment]
I suspect this is a taskwarrior-tui bug. Can you file it there, referencing this one?

**Comment by gms8994** [comment]
I'm seeing the same in task 3.4.1. 

```
news.version=3.4.1

data.location=**********

include /opt/homebrew/Cellar/task/3.4.1/share/doc/task/rc/solarized-dark-256.theme

alias.dailystatus=status:completed end.after:today all
alias.punt=modify wait:1d
alias.someday=mod +someday wait:someday

search.case.sensitive=no

uda.link.type=string
uda.link.label=URL

urgency.user.tag.Someday.coefficient=-5
calendar.holidays=full

sync.aws.region=**********
sync.aws.bucket=**********
sync.aws.access_key_id=**********
sync.aws.secret_access_key=**********
sync.encryption_secret=**********
```

The path to that theme does actually exist. `t show | grep ^color` shows a similar statement to https://github.com/GothenburgBitFactory/taskwarrior/issues/3461#issuecomment-2349090480. Changing the config to point to an existing file does indeed fail. Note that this is with iTerm2 on a mac. `task <id> start` does in fact show colors correctly, so I'm unsure if the problem is with the theme config or something else.

<img width="389" alt="Image" src="https://github.com/user-attachments/assets/0727c9ad-1777-426f-85ef-b12139a0bdf7" />

<img width="111" alt="Image" src="https://github.com/user-attachments/assets/c327ac88-b388-43b3-afbc-a4db6baf2b32" />

**Comment by djmitche** [comment]
What does `task color` show?

**Comment by Gambusia** [comment]
I'm having the same problem on Linux Mint 21.3. Everything seems to be properly configured, but the colour scheme is not rendering properly. The output of `task color` clearly shows that the terminal is capable of rendering colours properly. They all seem correct. The problem is the colour theme in the task interface.

**Comment by djmitche** [comment]
What do you mean by "task interface" -- can you give a screenshot?

**Comment by vitallish** [comment]
Hi all, just also chiming in with an issue. On Fedora 42 with kitty terminal. 

`task color` looks good and `task color legend` look good:

<img width="394" height="308" alt="Image" src="https://github.com/user-attachments/assets/dd9af37e-3d69-4626-ade9-4444508e223a" />

If I set a color manually in my taskrc, the color legend updates - so it looks like the config is working. happy to help debug!

**Comment by djmitche** [comment]
If I'm reading that correctly, everything's working. What's to debug?

**Comment by vitallish** [comment]
> If I'm reading that correctly, everything's working. What's to debug?

Sorry did so more checking and realized my due dates didn't transfer which is why i wasn't seeing color - apologies - I'll delete my comment

**Comment by djmitche** [comment]
Thanks @vitalish!

I'm going to close this issue as it's been several different people commenting with color-related issues, but without enough detail to debug, or followup to answer questions. If you, dear reader, are here with a color-theme problem, please file a new issue and read over the responses above to be sure you've included all of the relevant information!
